File: deob-ast/src/main/java/dev/openrs2/deob/ast/transform/UnencloseTransformer.java
package dev.openrs2.deob.ast.transform;

import com.github.javaparser.ast.CompilationUnit;
import com.github.javaparser.ast.Node;
import com.github.javaparser.ast.expr.EnclosedExpr;
import dev.openrs2.deob.ast.util.NodeUtils;

public final class UnencloseTransformer extends Transformer {
	@Override
	public void transform(CompilationUnit unit) {
		NodeUtils.walk(unit, Node.TreeTraversal.POSTORDER, EnclosedExpr.class, expr -> {
			expr.replace(expr.getInner().clone());
		});
	}
}

File: deob-ast/src/main/java/dev/openrs2/deob/ast/transform/UnencloseTransformer.kt
package dev.openrs2.deob.ast.transform

import com.github.javaparser.ast.CompilationUnit
import com.github.javaparser.ast.expr.EnclosedExpr
import dev.openrs2.deob.ast.util.walk
import javax.inject.Singleton

@Singleton
class UnencloseTransformer : Transformer() {
    override fun transformUnit(
        units: Map<String, CompilationUnit>,
        unit: CompilationUnit
    ) {
        unit.walk { expr: EnclosedExpr ->
            expr.replace(expr.inner.clone())
        }
    }
}
